ssc/Pos/Ssc/Base.hs

```diff
@@ -100,32 +108,50 @@ mkSignedCommitment
     => ProtocolMagic -> SecretKey -> EpochIndex -> Commitment -> SignedCommitment
 mkSignedCommitment pm sk i c = (toPublic sk, c, sign pm SignCommitment sk (i, c))
 
-toLocalSlotIndex :: HasProtocolConstants => SlotCount -> LocalSlotIndex
-toLocalSlotIndex = unsafeMkLocalSlotIndex . fromIntegral
+toLocalSlotIndex :: ProtocolConstants -> SlotCount -> LocalSlotIndex
+toLocalSlotIndex pc = unsafeMkLocalSlotIndexExplicit pc . fromIntegral
+
+isCommitmentIdxExplicit :: ProtocolConstants -> LocalSlotIndex -> Bool
+isCommitmentIdxExplicit pc =
+    inRange (toLocalSlotIndex pc 0,
+             toLocalSlotIndex pc (pcSlotSecurityParam pc - 1))
 
 isCommitmentIdx :: HasProtocolConstants => LocalSlotIndex -> Bool
-isCommitmentIdx =
-    inRange (toLocalSlotIndex 0,
-             toLocalSlotIndex (slotSecurityParam - 1))
+isCommitmentIdx = isCommitmentIdxExplicit protocolConstants
+
+isOpeningIdxExplicit :: ProtocolConstants -> LocalSlotIndex -> Bool
+isOpeningIdxExplicit pc =
+    inRange (toLocalSlotIndex pc (2 * pcSlotSecurityParam pc),
+             toLocalSlotIndex pc (3 * pcSlotSecurityParam pc - 1))
 
 isOpeningIdx :: HasProtocolConstants => LocalSlotIndex -> Bool
-isOpeningIdx =
-    inRange (toLocalSlotIndex (2 * slotSecurityParam),
-             toLocalSlotIndex (3 * slotSecurityParam - 1))
+isOpeningIdx = isOpeningIdxExplicit protocolConstants
+
+isSharesIdxExplicit :: ProtocolConstants -> LocalSlotIndex -> Bool
+isSharesIdxExplicit pc =
+    inRange (toLocalSlotIndex pc (4 * pcSlotSecurityParam pc),
+             toLocalSlotIndex pc (5 * pcSlotSecurityParam pc - 1))
```

Comment:
I know this was in the original code, but any idea what these magic numbers (`4` and 5`) are about?